public/app/features/scenes/scenes/cache.tsx

```diff
@@ -0,0 +1,149 @@
+import { SceneVariableSet } from '../variables/sets/SceneVariableSet';
+import { CustomVariable } from '../variables/variants/CustomVariable';
+import { QueryVariable } from '../variables/variants/query/QueryVariable';
+
+export function getCacheTest(standalone: boolean): Scene {
+  const getVariables = () =>
+    new SceneVariableSet({
+      variables: [
+        new CustomVariable({
+          name: 'metricCached',
+          cacheKey: 'metric',
+          query: 'job : job, instance : instance',
+        }),
+        new QueryVariable({
+          name: 'selectedMetricVarCached',
+          cacheKey: 'selectedMetricVarCached',
+          query: { query: 'label_values(go_gc_duration_seconds,${metricCached})' },
+          datasource: { uid: 'gdev-prometheus', type: 'prometheus' },
+        }),
+        new QueryVariable({
+          name: 'instance (will requery on layout toggle)',
+          query: { query: 'label_values(go_gc_duration_seconds, instance)' },
+          datasource: { uid: 'gdev-prometheus', type: 'prometheus' },
+        }),
+      ],
+    });
+
+  const getSubMenu = () =>
+    new SceneSubMenu({
+      children: [
+        new VariableValueSelectors({}),
+        new SceneSubMenuSpacer(),
+        sceneToggle,
+        new SceneTimePicker({ isOnCanvas: true }),
+      ],
+    });
+
+  const getLayoutKids = (l: string) => {
+    return [
+      new SceneFlexLayout({
+        direction: 'row',
+        children: [
+          new SceneCanvasText({ text: `${l}: With initial count state`, size: { width: 300 } }),
+          new TestSceneObject({ cacheKey: 'test' }),
+          new TestSceneObject({}),
+        ],
+      }),
+
+      new SceneFlexLayout({
+        direction: 'row',
+        children: [
+          new SceneCanvasText({ text: `${l}: Without  initial count state`, size: { width: 300 } }),
+          new TestSceneObject({ cacheKey: 'test1', count: 5 }),
+          new TestSceneObject({ count: 5 }),
+        ],
+      }),
+    ];
+  };
+
+  const layout = new SceneFlexLayout({
+    direction: 'column',
+    children: getLayoutKids('Layout 1'),
+  });
+
+  const sceneToggle = new SceneRadioToggle({
+    options: [
+      { value: 'l1', label: 'Layout 1' },
+      { value: 'l2', label: 'Layout 2' },
+    ],
+    value: 'l1',
+    onChange: (value) => {
+      const vars = getVariables();
+      if (value === 'l1') {
+        scene.setState({ $variables: vars, subMenu: getSubMenu() });
+        // Activating manually, as we are not creating a new scene in this demo
+        vars.activate();
+        layout.setState({
+          children: getLayoutKids('Layout 1'),
+        });
+      } else {
+        scene.setState({ $variables: vars, subMenu: getSubMenu() });
+        // Activating manually, as we are not creating a new scene in this demo
+        vars.activate();
+
+        layout.setState({ children: getLayoutKids('Layout 2') });
+      }
+    },
+  });
```

Comment:
@torkelo  - this demo is very hacky as I'm not creating two separate Scenes. And the `activate` call here is only because of that - the updated variables are not getting activated obviously, as there is no Scene.activate called :) I'll simplify this demo.